sql/parser/eval.go

```diff
@@ -428,6 +428,14 @@ func init() {
 	cmpOps[cmpArgs{In, tupleType, tupleType}] = evalTupleIN
 }
 
+// EvalContext defines the context in which to evaluate an expression, allowing
+// the retrieval of state such as the node ID or statement start time.
+type EvalContext struct {
```

Comment:
@vivekmenezes This will be of interest to your `current_timestamp` work. I think you'll just need to add another field here, populate it when the context is created and use it from the `current_timestamp` function.